openassessment/xblock/submission_mixin.py

```diff
@@ -370,6 +370,29 @@ def remove_all_uploaded_files(self, data, suffix=''):  # pylint: disable=unused-
                 break
         return {'success': True, 'removed_num': removed_num}
 
+    @XBlock.json_handler
+    def remove_uploaded_file(self, data, suffix=''):  # pylint: disable=unused-argument
+        """
+        Removes uploaded user file.
+        """
+        filenum = data.get('filenum', -1)
+        try:
+            filenum = int(filenum)
+        except ValueError:
+            filenum = -1
+
+        removed = file_upload_api.remove_file(self._get_student_item_key(filenum))
+        if removed:
+            saved_file_descriptions = json.loads(self.saved_files_descriptions)
+            saved_file_descriptions[filenum] = None
+            self.saved_files_descriptions = json.dumps(saved_file_descriptions)
+
+            saved_files_names = json.loads(self.saved_files_names)
+            saved_files_names[filenum] = None
```

Comment:
It appears that we are simply assigning the value of None to the array position that corresponds to the removed file index. However, it would be better to remote that element from the array altogether:

```
saved_file_descriptions.pop(filenum)
saved_files_names.pop(filenum)

```

Reply:
We would then have to adjust the URLs of all of the remaining elements, replacing it with None lets us just pass over it and ignore it without having to modify the other files in any way